### PR TITLE
fix get ip at runtime for redirect address and automation scripts

### PR DESCRIPTION
**Pull request by AIIX**
- Build qr code redirect url by getting the local ip at runtime

### CODE REVIEW

File: ovos_PHAL_plugin_oauth/__init__.py [hunk 1]
```diff
@@ -253,7 +253,8 @@ def generate_qr(self, url, skill_id, app_id):
         return f"{temp_dir}/{skill_id}_{app_id}_oauth_qr_{img_id}.png"
 
     def build_plugin_service_url(self, oauth_url, skill_id, app_id, client_id=None, client_secret=None):
-        redirect_uri = f"http://{self.local_flask_host}:{self.port}/auth/callback/{skill_id}_{app_id}"
```
Comment (NeonDaniel) [inline ovos_PHAL_plugin_oauth/__init__.py:256]: Can `self.local_flask_host` be removed completely, or should it be updated here?

File: .github/workflows/license_tests.yml [hunk 2]
```diff
@@ -1,11 +1,6 @@
 name: Run License Tests
 on:
   push:
-    branches:
```
Comment (NeonDaniel) [inline .github/workflows/license_tests.yml]: Is this change intentional? This will make test run on any `push` event

**Review by NeonDaniel — CHANGES_REQUESTED** [review]

**Comment by NeonDaniel** [comment]
For failing license tests, `ovos-PHAL-plugin-oauth` should be excluded in the automation: https://github.com/OpenVoiceOS/ovos-PHAL-plugin-oauth/blob/master/.github/workflows/license_tests.yml#L40

`exclude: '^(tqdm|ovos-PHAL-plugin-oauth).*'`

File: .github/workflows/license_tests.yml (hunk 2 above)
Comment (AIIX) [reply .github/workflows/license_tests.yml]: its a copy paste from another phal plugin

**Review by NeonDaniel — APPROVED** [review]
